tests/test_series.py

```diff
@@ -1078,25 +1078,94 @@ def test_types_groupby_agg() -> None:
         r"The provided callable <built-in function (min|sum)> is currently using",
         upper="2.2.99",
     ):
-        check(assert_type(s.groupby(level=0).agg(sum), pd.Series), pd.Series)
+
+        def sum_sr(s: pd.Series[int]) -> int:
+            # type of `sum` not well inferred by mypy
+            return s.sum()
+
+        check(
+            assert_type(s.groupby(level=0).agg(sum_sr), "pd.Series[int]"),
+            pd.Series,
+            np.integer,
+        )
         check(
             assert_type(s.groupby(level=0).agg([min, sum]), pd.DataFrame), pd.DataFrame
         )
 
 
+def test_types_groupby_transform() -> None:
```

Comment:
I think you should add tests for  two of the string transform arguments (e.g., "mean", "first")